channels/smartcard/client/smartcard_pack.c

```diff
@@ -409,10 +417,11 @@ LONG smartcard_unpack_redir_scard_handle_ref(SMARTCARD_DEVICE* smartcard, wStrea
 LONG smartcard_pack_redir_scard_handle_ref(SMARTCARD_DEVICE* smartcard, wStream* s,
         REDIR_SCARDHANDLE* handle)
 {
-	Stream_Write_UINT32(s, handle->cbHandle); /* Length (4 bytes) */
```

Comment:
Are you sure you can remove the lenght field if there is no data?

Reply:
> The image you committed is corrupted and can not be read.

I can visualize this docs/Smartcard-Logon-Dataflow.png on firefox or safari from github.
But I will make it a graphviz dot file, so it’ll be textual and easier to maintain.